python/feldera/pipeline.py

```diff
@@ -344,3 +344,49 @@ def get(name: str, client: FelderaClient) -> 'Pipeline':
         except FelderaAPIError as err:
             if err.status_code == 404:
                 raise RuntimeError(f"Pipeline with name {name} not found")
+
+    def query(self, query: str) -> Generator[Mapping[str, Any], None, None]:
+        """
+        Executes an ad-hoc SQL query on this pipeline and returns the result in the specified format.
+
+        :param query: The SQL query to be executed.
+        :return: A generator that yields the rows of the result as Python dictionaries.
+        """
+
+        if self.status() not in [
```

Comment:
This check may be redundant, since the pipeline manager should return the same error if the pipeline is not running. Can we check that this is the case (and the user ends up getting a meaningful error message)? Same goes for the other two methods.

Reply:
Yeah, the checks are redundant, but the older commit missed HTTP errors for streaming responses. 

The latest commit fixes that.